packages/ui/src/views/secrets/new-secret/new-secret-form.tsx

```diff
@@ -0,0 +1,137 @@
+import { useForm, type SubmitHandler } from 'react-hook-form'
+
+import {
+  Accordion,
+  Button,
+  ButtonGroup,
+  ControlGroup,
+  Fieldset,
+  FormWrapper,
+  Input,
+  Spacer,
+  Textarea
+} from '@/components'
+import { SandboxLayout, TranslationStore } from '@/views'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { z } from 'zod'
+
+const newSecretformSchema = z.object({
+  name: z.string().min(1, { message: 'Please provide a name' }),
+  password: z.string().min(1, { message: 'Please provide a password' }),
+  description: z.string().optional(),
+  tags: z.string().optional()
+})
+
+export type NewSecretFormFields = z.infer<typeof newSecretformSchema> // Automatically generate a type from the schema
+
+interface NewSecretPageProps {
```

Comment:
let's call it `CreateSecret` everywhere.. "new" can be confused with versioning (vs "old")